splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/SparkExplainIT.java

```diff
@@ -805,11 +805,26 @@ public void testCrossJoinBroadcastRight() throws Exception {
     }
 
     @Test
-    public void testCrossJoinBroadcastLeft() throws Exception {
+    public void testCrossJoinNeverBroadcastLeft() throws Exception {
+        /* DB-9579
+         * In case of a cross join, we only check if the right side can be broadcast and never broadcast the left
+         * side. Reason is that when the plan reaches Spark and it doesn't have a sort node, it might be the case
+         * that there are sort keys but the sort node is optimized away because the left side row order satisfies
+         * the sort keys. If we broadcast the left side, resulting row order might be different.
+         */
         if (useSpark.equalsIgnoreCase("false")) return; // cross join only has Spark implementation
         String sqlText = "sparkexplain select count(*) from --splice-properties joinOrder=fixed\n" +
                 "        big --splice-properties useSpark=true, joinStrategy=cross\n" +
                 "        inner join t1 on 1=1";
-        testQueryContains(sqlText, "BroadcastNestedLoopJoin BuildLeft, Cross", methodWatcher, true);
+        testQueryContains(sqlText, "CartesianProduct", methodWatcher, true);
+    }
+
+    @Test
+    public void testCrossJoinBroadcastRightNoHint() throws Exception {
+        if (useSpark.equalsIgnoreCase("false")) return; // cross join only has Spark implementation
+        String sqlText = "sparkexplain select count(*) from\n" +
+                "        t1 --splice-properties useSpark=true\n" +
+                "        inner join big on 1=1";
+        testQueryContains(sqlText, "BroadcastNestedLoopJoin BuildRight, Cross", methodWatcher, true);
```

Comment:
Could you also add a check for static explain whether the table `BIG` is the picked as the left table of the cross join? Thanks!

Reply:
Also, please add an IT with data to test the preservation of sortorder for eligible cross join cases. Thanks!

Reply:
> Could you also add a check for static explain whether the table `BIG` is the picked as the left table of the cross join? Thanks!

Done.

Reply:
> Also, please add an IT with data to test the preservation of sortorder for eligible cross join cases. Thanks!

Done. I add this one to CrossJoinIT.